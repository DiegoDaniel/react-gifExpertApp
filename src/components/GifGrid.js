import React from 'react' //, { useEffect, useState }
import { useFetchGifs } from '../hooks/useFetchGifs'
import { GifGridItem } from './GifGridItem';
import PropTypes from 'prop-types';

export const GifGrid = ( { category } ) => {
            //renombro
    const { data:images, loading } = useFetchGifs( category );

    return (
        <>
            <h3>{ category }</h3>
            { loading && <p className="animate__animated animate__flash">Cargando...</p>}
            <div className="card-grid">
                { 
                images.map( ( img ) => {
                    return <GifGridItem 
                        key={ img.id }
                        //Se puede hacer spread de CADA UNO de los valores de img
                        //Para enviar al componente, en lugar de hacer id={img.id},, title={img.title},
                        //El spread devuelve todos los valores del objeto.
                        { ...img } />;
                })
                }
            </div>
        </>
    )
}

GifGrid.propTypes = {
    category: PropTypes.string.isRequired
}