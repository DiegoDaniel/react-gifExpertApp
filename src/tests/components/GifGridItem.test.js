import React from 'react';
import { shallow } from 'enzyme';
import { GifGridItem } from '../../components/GifGridItem';


describe('Pruebas en <GifGridItem />', () => {
    
    const url = 'google.com';
    const title = 'Diego PNG';
    const wrapper = shallow( <GifGridItem url={ url } title={ title }/> );
    
    
    test('Debe renderizar correctamente el componente ', () => {

        expect( wrapper ).toMatchSnapshot();

    });

    test('debe tener un pÃ¡rrafo con el titulo', () => {
        
        const p = wrapper.find('p');

        expect( p.text().trim() ).toBe( title );


    });

    test('La imagen debe tener source y alt', () => {

        const img = wrapper.find('img');
        //2 formas de ver las props del componente img.
        //console.log(img.props().src );
        //console.log(img.prop('src') );

        expect( img.prop('src') ).toBe( url );
        expect( img.prop('alt') ).toBe( title );
    });
    
    test('debe tener la clase fadeIn', () => {
        
        const className = 'animate__fadeIn';

        const div = wrapper.find('div');

        expect(div.prop('className')).toContain( className );

    })
    
    
    
})
