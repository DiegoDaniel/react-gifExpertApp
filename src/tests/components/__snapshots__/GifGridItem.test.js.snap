// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGridItem /> Debe renderizar correctamente el componente  1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Diego PNG"
    src="google.com"
  />
  <p>
    Diego PNG
  </p>
</div>
`;
